build_and_release.rb

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/ruby
+
+# This script builds the code, overriding the version number of the pom with
+# one of the form MAJOR.MINOR.BUILD. The major and minor version numbers are
+# taken from the pom file (see the BLUEFLOOD_VERSION maven parameter). The
+# build number should be passed in as the first and only command line argument
+# to the script. The script will then turn a snapshot version string, such as
+# "2.0-SNAPSHOT", into something suitable for a release build, like "2.0.36".
+#
+# Running maven from the command line (e.g. `mvn package`), instead of the
+# script, will still produce artifacts with the SNAPSHOT version string.
+
+require 'rexml/document'
+
+
+def get_version_number(build_number)
+  pom = REXML::Document.new(File.new("pom.xml"))
+  version = pom.elements["project/properties/BLUEFLOOD_VERSION"].text
+
+  if version.end_with? '${VERSION_SUFFIX}'
```

Comment:
@izrik I was thinking from outside the script. The script takes a maven project with x.x.x-SNAPSHOT as version and produces artifacts with x.x.${BUILD_NUMBER}. What if the version of the project is not SNAPSHOT in the first place? Shouldnt it fail? Its just a precaution when someone manually changes BLUEFLOOD_VERSION to not have -SNAPHOT and tried to run this script. For ex: If someone cuts a release on project using release plugin, I think it would fail if the underlying project does not have SNAPSHOT version.